ceph/utils.py

```diff
@@ -358,6 +361,38 @@ def create_ceph_nodes(
     return ceph_nodes
 
 
+def find_networks(cloud, subnets):
+    """Return cloud network id names.
+
+    Use logical subnet names defined in global conf to fetch
+     cloud network names using public_networks + cluster_networks.
+
+    If logical subnet names found in public_networks + cluster_networks,
+     then return corresponding cloud network names.
+    else if not found, then find new cloud network names.
+
+    Args:
+        cloud: cloud object
+        subnets: list of logical subnet names
+
+    Returns:
+        cloud_network_names
+    """
+    cloud_network_names = []
+    for net in subnets:
+        if net in networks:
+            cloud_network_names.append(networks[net])
+        else:
+            pre_nwlist = list(
+                set(cloud.default_network_names)
+                - set([i.name for i in networks.values()])
+            )
+            subnet = cloud.get_network(subnet_list=pre_nwlist)
+            networks.update({net: subnet[0]})
+            cloud_network_names.append(networks[net])
+    return cloud_network_names
+
+
```

Comment:
Is it possible to avoid this method.

IMO, the network names can be determine within the compute module. This allows us to use compute specific API's to determine the `ids` or `names` of the network.

Reply:
Removed.